### PR TITLE
Add authorized keys.d support

### PR DESCRIPTION
**Pull request by fortinj66**
Add Initial authorized_keys.d support.  
(Note this also includes patched to allow authorized_keys.d to function and the vsphere fix for hostnames...  This is of course be removed for a pull to openshift/machine-config-operator)

### CODE REVIEW

File: pkg/daemon/update.go [hunk 1]
```diff
@@ -1725,8 +1725,28 @@ func getFileOwnership(file ign3types.File) (int, int, error) {
 }
 
 func (dn *Daemon) atomicallyWriteSSHKey(keys string) error {
-	authKeyPath := filepath.Join(coreUserSSHPath, "authorized_keys")
+	var authKeyPath string
 
+	// If we are using FCOS then use /home/core/.ssh/authorized_keys.d/ignition
+    if dn.os.IsFCOS() {
```
Comment (vrutkovs) [inline pkg/daemon/update.go:1731]: mix of tabs and spaces

File: pkg/daemon/update.go [hunk 2]
```diff
@@ -1725,8 +1725,28 @@ func getFileOwnership(file ign3types.File) (int, int, error) {
...
+		authKeyPath = filepath.Join(coreUserSSHPath, "authorized_keys.d/authorized_keys_slice")
+		authKeyPathIgnite := filepath.Join(coreUserSSHPath, "authorized_keys.d/ignition")
+		authKeyPathDefault := filepath.Join(coreUserSSHPath, "authorized_keys")
+
+		//First time rename authorized_keys.d/ignition to authorized_keys.d/authorized_keys_slice
```
Comment (vrutkovs) [inline pkg/daemon/update.go:1736]: Why not use `/home/core/.ssh/authorized_keys`?

File: templates/common/vsphere/units/vsphere-hostname.service.yaml [hunk 3]
```diff
@@ -7,7 +7,7 @@ contents: |
   After=vmtoolsd.service
 
   # Only run on first boot and only if the host does not have a name yet
-  ConditionPathExists=/etc/ignition-machine-config-encapsulated.json
```
Comment (vrutkovs) [inline templates/common/vsphere/units/vsphere-hostname.service.yaml]: Needs to be a separate commit.

I don't think we need to check for `/etc/ignition-machine-config-encapsulated.json` at all - checking for `/etc/hostname` should be sufficient (nothing writes on FCOS afaik and users can redefine /etc/hostname)

File: pkg/daemon/update.go (hunk 2 above)
Comment (fortinj66) [reply pkg/daemon/update.go:1736]: To me that defeats the purpose of the PR which is to enable authorized_keys.d/ usage.  This negates the need for Christian's security PR, so ssh works out of the box.  I do wish that the ignition process hadn't named the file `ignition` though.  

Although I do see the alternative to use authorized_keys and just delete authorized_keys.d.  It would simplify the code.

I'm not passionate around either as long as "something: is done...

File: templates/common/vsphere/units/vsphere-hostname.service.yaml (hunk 3 above)
Comment (fortinj66) [reply templates/common/vsphere/units/vsphere-hostname.service.yaml]: Agreed about the seperate commit.  Unfortunately, this was how I was able to test.  If we go forward I'll seperate them.

Having the check is very slightly cleaner as it wouldn't run on first boot, but getting rid of it is fine to me also...

File: pkg/daemon/update.go (hunk 1 above)
Comment (fortinj66) [reply pkg/daemon/update.go:1731]: fixed
